internal/libs/autofile/group.go

```diff
@@ -88,7 +90,12 @@ func OpenGroup(headPath string, groupOptions ...func(*Group)) (*Group, error) {
 		return nil, err
 	}
 
+	if logger == nil {
```

Comment:
It looks as if the only place we use this function outside the package is in internal/consensus/wal.go, so I think we could just mandate it is not nil and let it panic if someone tries to cheat (as with contexts)